enterprise/frontend/src/embedding-sdk/hooks/private/use-load-question.ts

```diff
@@ -74,20 +75,24 @@ export function useLoadQuestion({
     deferredRef.current?.resolve();
   });
 
+  // Avoid re-running the query if the parameters haven't changed.
+  const sqlParameterKey = JSON.stringify(initialSqlParameters);
```

Comment:
When passing `initialSqlParameters={{ID: "12"}}`, this may cause an infinite re-render because the object reference is created every render. This makes it so that we only update the `loadQuestion` callback's reference if the SQL parameter actually changed.

Reply:
@WiNloSt Good call, made it key-order independent in https://github.com/metabase/metabase/pull/50728/commits/171d2a525825e5c310d7eb3de3be3806de4b313d.

PS. I went with creating a dependency key that is key-order independent instead of using isEqual, as that would require using `usePrevious` and `isEqual` to compare other dependencies as well.